x-pack/plugins/ingest_manager/server/services/datasource.ts

```diff
@@ -145,6 +145,7 @@ class DatasourceService {
   public async delete(
     soClient: SavedObjectsClientContract,
     ids: string[],
+    unassignFromAgentConfigs: boolean = true,
```

Comment:
I think this should go into the options parameter.

We can still default it to true/whatever and it's backwards compatible.

This reminded me of https://ariya.io/2011/08/hall-of-api-shame-boolean-trap

Reply:
adjusted